Backend/main.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/gorilla/mux"
+)
+
+var (
+	connection = connectToRedis()
+	ctx        = context.Background()
+)
+
+type Image struct {
+	URL string `json:"url"`
+	GCP string `json:"GCP"`
+}
+
+func connectToRedis() *redis.Client {
+
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     os.Getenv("REDIS_URL"),
+		Password: os.Getenv("REDIS_PASSWORD"),
+		DB:       0,
+	})
+
+	if err := rdb.Ping(ctx).Err(); err != nil {
+		return nil
+	}
+
+	return rdb
+}
+func insertToDB(key string, value string) error {
+	_, err := connection.Set(ctx, key, value, 0).Result()
+	if err != nil {
+		fmt.Println(err)
+	}
+	return err
+}
+func getFromDB(key string) (interface{}, error) {
+	//val, err := connection.Do(ctx, "GET", key).Result()
+	val, err := connection.Get(ctx, key).Result()
+	if err != nil {
+		fmt.Println(err)
+	}
+	return val, err //fmt.Sprintf("%s", val)
+}
+
+func allKeys() []string {
```

Comment:
What `allKeys` do you mean? Can you make the name of this function more precisely?